tools/PI/DevHome.PI/Helpers/WindowHelper.cs

```diff
@@ -410,6 +431,26 @@ internal static unsafe uint GetProcessIdFromWindow(HWND hWnd)
         return processID;
     }
 
+    internal static void TranslateUWPProcess(HWND hWnd, ref Process process)
+    {
+        if (process.ProcessName.Equals("ApplicationFrameHost", StringComparison.OrdinalIgnoreCase))
+        {
+            var processId = GetProcessIdFromUWPWindow(hWnd);
+            if (processId != 0)
+            {
+                process = Process.GetProcessById((int)processId);
+            }
+        }
+    }
+
+    internal static unsafe uint GetProcessIdFromUWPWindow(HWND hWnd)
+    {
+        uint processId = 0;
+        LPARAM lParam = new((nint)(&processId));

```

Comment:
In an effort to reduce unsafe code, what do you think about something like this instead?

```
internal static uint GetProcessIdFromUWPWindow(HWND hWnd)
{
    UWPProcessFinder processFinder = new();

    PInvoke.EnumChildWindows(hWnd, processFinder.EnumChildProc, IntPtr.Zero);
    return processFinder.UWPProcessId;
}

private sealed class UWPProcessFinder
{
    public uint UWPProcessId { get; set; }

    public BOOL EnumChildProc(HWND hWnd, LPARAM data)
    {
        unsafe
        {
            var className = stackalloc char[256];
            var classNameLength = PInvoke.GetClassName(hWnd, className, 256);
            if (classNameLength > 0)
            {
                string classNameString = new(className, 0, classNameLength);
                if (classNameString.StartsWith("Windows.UI.Core.", StringComparison.OrdinalIgnoreCase))
                {
                    uint hWndProcessId = 0;
                    _ = PInvoke.GetWindowThreadProcessId(hWnd, &hWndProcessId);
                    UWPProcessId = hWndProcessId;
                    return false;
                }
            }
        }

        return true;
    }
}

```

Reply:
Nice!
I have reduced the unsafe code. I have just declared the proc as unsafe instead of using the unsafe block to reduce nesting 😃